compiler/luci/pass/src/ResolveCustomOpMaxPoolWithArgmaxPass.cpp

```diff
@@ -0,0 +1,774 @@
+
+  // define Padding
+  auto padding = string_to_padding(padding_param);
+
+  // define Filter
+  luci::Filter filter;
+  filter.h(ksize_param[1]);
+  filter.w(ksize_param[2]);
+
+  // define Stride
+  luci::Stride stride;
+  stride.h(strides_param[1]);
+  stride.w(strides_param[2]);
+
+  // Create MaxPool
+  auto maxpool = graph->nodes()->create<luci::CircleMaxPool2D>();
+  {
+    maxpool->name(name + "/MaxPool2D");
+    luci::add_origin(maxpool, origin);
+
+    maxpool->stride()->h(stride.h());
+    maxpool->stride()->w(stride.w());
+
+    maxpool->filter()->h(filter.h());
+    maxpool->filter()->w(filter.w());
+
+    maxpool->padding(padding);
+
+    maxpool->fusedActivationFunction(luci::FusedActFunc::NONE);
+
+    // dtype of first output of MaxPoolWithArgmax is same as input's
+    maxpool->dtype(input->dtype());
+    define_max_pool_shape(input, maxpool);
+
+    maxpool->value(input);
+  }
+
+  // Create Split
+  auto split = graph->nodes()->create<luci::CircleSplit>();
+  {
+    split->name(name + "/Split");
+    luci::add_origin(split, origin);
+
+    // Create split_dim
+    auto split_dim = create_scalar<loco::DataType::S32>(graph, depth_dimension);
+    {
+      split_dim->name(split->name() + "/Dim");
+      luci::add_origin(split_dim, origin);
+    }
+
+    // Split input feature map to FM's with depth equal to 1
+    split->dtype(input->dtype());
+    copy_shape(input, split);
+    split->num_split(int32_t(input_depth));
+
+    split->split_dim(split_dim);
+    split->input(input);
+  }
+
+  // Create Concatenation
+  auto concat = graph->nodes()->create<luci::CircleConcatenation>(input_depth);
+  {
+    concat->name(name + "/Concatenation");
+    luci::add_origin(concat, origin);
+
+    concat->fusedActivationFunction(luci::FusedActFunc::NONE);
+
+    // Output of branches is FLOAT32
+    concat->dtype(loco::DataType::FLOAT32);
+    copy_shape(maxpool, concat);
+
+    concat->axis(depth_dimension);
+  }
+
+  /**
+   * Note: we need define idx from input_tensor of maximum element in MaxPool's sliding window.
+   * For this we split input tensor by channels, define idx in sliding window and convert this idx
+   * to idx from source input_tensor using FloorDiv, Mul and Add operations with constant tensors.
+   */
+  // create branches
+  for (uint32_t br_n = 0; br_n < input_depth; ++br_n)
+  {
+    // Create CircleSplitOut
+    auto split_out = graph->nodes()->create<luci::CircleSplitOut>();
+    {
+      split_out->name(name + "/SplitOut_" + std::to_string(br_n));
+      luci::add_origin(split_out, origin);
+
+      split_out->index(br_n);
+      split_out->dtype(split->dtype());
+
+      // shape of split out equal to shape of split except split_dim, which equal to 1
+      copy_shape(split, split_out);
+      split_out->dim(depth_dimension).set(1);
+
+      split_out->input(split);
+    }
+
+    // Define idx of max element in Window:
+    // Create Conv2D
+    auto conv = graph->nodes()->create<luci::CircleConv2D>();
```

Comment:
This conv implements a special "space_to_depth" operation with arbitrary block width and height, you can use this analogy to describe this convolution purpose.